api/rds.proto

```diff
@@ -484,80 +497,126 @@ message RateLimit {
   // .. note::
   //
   //   The filter supports a range of 0 - 10 inclusively for stage numbers.
-  google.protobuf.UInt32Value stage = 1;
+  google.protobuf.UInt32Value stage = 1 [(validate.rules).uint32.lte = 10];
 
   // The key to be set in runtime to disable this rate limit configuration.
   string disable_key = 2;
 
   message Action {
     // The following descriptor entry is appended to the descriptor:
-    // ("source_cluster", "<local service cluster>")
-    message SourceCluster {
-    }
+    //
+    // .. code-block:: cpp
+    //
+    //   ("source_cluster", "<local service cluster>")
+    //
+    // <local service cluster> is derived from the :option:`--service-cluster` option.
```

Comment:
Unrelated, but were we planning on essentially deprecating --service-cluster and --service-node in exchange for providing a Locality or Metadata bootstrap option that generalizes? The format for that CLI option would be a text proto or JSON.

Reply:
Yeah agreed that should be the way we go. Not sure if we have a ticket tracking this. I will look.